### PR TITLE
Add governance support to LiquidityProtection

### PR DESCRIPTION
**Pull request by lbeder**

### CODE REVIEW

File: solidity/contracts/liquidity-protection/LiquidityProtection.sol [hunk 1]
```diff
@@ -476,11 +448,11 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
             "ERR_PROTECTIONS_MISMATCH"
         );
 
-        // burn the governance tokens from the caller
-        govToken.destroy(
-            msg.sender,
-            liquidity1.reserveToken == networkToken ? liquidity1.reserveAmount : liquidity2.reserveAmount
-        );
+        // burn the governance tokens from the caller. we need to transfer the tokens to the contract itself, since only
+        // token owners can burn their tokens
```
Comment (yudilevi) [inline solidity/contracts/liquidity-protection/LiquidityProtection.sol:452]: maybe owners -> holders?

File: solidity/contracts/liquidity-protection/LiquidityProtection.sol [hunk 2]
```diff
@@ -779,9 +753,11 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         // add the pool tokens to the system
         store.incSystemBalance(liquidity.poolToken, liquidity.poolAmount);
 
-        // if removing network token liquidity, burn the governance tokens from the caller
+        // if removing network token liquidity, burn the governance tokens from the caller. we need to transfer the
+        // tokens to the contract itself, since only token owners can burn their tokens
```
Comment (yudilevi) [inline solidity/contracts/liquidity-protection/LiquidityProtection.sol:757]: maybe owners -> holders?

File: solidity/contracts/liquidity-protection/LiquidityProtection.sol (hunk 1 above)
Comment (lbeder) [reply solidity/contracts/liquidity-protection/LiquidityProtection.sol:452]: 👍

File: solidity/contracts/liquidity-protection/LiquidityProtection.sol (hunk 2 above)
Comment (lbeder) [reply solidity/contracts/liquidity-protection/LiquidityProtection.sol:757]: 👍

File: solidity/test/LiquidityProtection.js [hunk 3]
```diff
@@ -1230,13 +1254,14 @@ contract('LiquidityProtection', (accounts) => {
                 const prevBalance = await getBalance(baseToken, baseTokenAddress, owner);
                 const prevGovBalance = await govToken.balanceOf(owner);
 
-                const res = await liquidityProtection.removeLiquidity(protectionIds[0], PPM_RESOLUTION);
+                const res = await govToken.approve(liquidityProtection.address, protection.reserveAmount);
```
Comment (yudilevi) [inline solidity/test/LiquidityProtection.js:1257]: only needed when removing network token liquidity

File: solidity/test/LiquidityProtection.js [hunk 4]
```diff
@@ -1287,7 +1312,12 @@ contract('LiquidityProtection', (accounts) => {
                 const prevBalance = await getBalance(baseToken, baseTokenAddress, owner);
                 const prevGovBalance = await govToken.balanceOf(owner);
 
-                const res = await liquidityProtection.removeLiquidity(protectionId, 800000);
+                const portion = new BN(800000);
+                const res = await govToken.approve(
```
Comment (yudilevi) [inline solidity/test/LiquidityProtection.js:1316]: only needed when removing network token liquidity

File: solidity/test/LiquidityProtection.js (hunk 3 above)
Comment (lbeder) [reply solidity/test/LiquidityProtection.js:1255]: 👍

**Review by drumenbassik — APPROVED** [review]

**Pull request merged**